packages/grafana-ui/src/components/Combobox/MultiCombobox.tsx

```diff
@@ -171,18 +211,18 @@ export const MultiCombobox = <T extends string | number>(props: MultiComboboxPro
               key={`${item.value}${index}`}
               {...getSelectedItemProps({ selectedItem: item, index })}
             >
-              {itemToString(item)}
+              {item.value === ALL_OPTION_VALUE ? allOptionItem.label : itemToString(item)}
```

Comment:
```suggestion
              {itemToString(item)}
```
Verify that this is needed